### PR TITLE
Extend fake policy backend for OOP adapter integration test

### PR DESCRIPTION
**Pull request by bianpengyuan**
This is a prep pr for adding out of process policy adapters integration. It extends fake policy backend to support [keyval template](https://github.com/istio/istio/tree/master/mixer/test/keyval) and adapter config. Also add code to generate oop handler config for the tests.

### CODE REVIEW

File: pkg/test/fakes/policy/backend.go [hunk 1]
```diff
@@ -224,11 +237,44 @@ func (b *Backend) HandleCheckNothing(ctx context.Context, req *checknothing.Hand
 	}
 
 	scope.Infof("Backend.HandleCheckNothing => OK")
+	validDuration := 5 * time.Second
```
Comment (ozevren) [inline pkg/test/fakes/policy/backend.go]: Nit: You can make these configurable. PolicyBackend has a control connection for controlling its behavior.

Comment (bianpengyuan) [reply pkg/test/fakes/policy/backend.go]: Done.

**Comment by bianpengyuan** [comment]
@ozevren please take another look, Thanks!

**Pull request merged**